flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/casting/ArrayToStringCastRule.java

```diff
@@ -91,74 +121,94 @@ protected String generateCodeBlockInternal(
         context.declareClassField(
                 className(StringBuilder.class), builderTerm, constructorCall(StringBuilder.class));
 
-        return new CastRuleUtils.CodeWriter()
-                .stmt(methodCall(builderTerm, "setLength", 0))
-                .stmt(methodCall(builderTerm, "append", strLiteral("[")))
-                .forStmt(
-                        methodCall(inputTerm, "size"),
-                        (indexTerm, loopBodyWriter) -> {
-                            String elementTerm = newName("element");
-                            String elementIsNullTerm = newName("elementIsNull");
+        final String resultStringTerm = newName("resultString");
+        final int precision = LogicalTypeChecks.getLength(targetLogicalType);
```

Comment:
nit: `precision` -> `length` here and at other locations

Reply:
Imho, that is confusing, because it's called `precision` on the type (CHAR and VARCHAR) and `length` should be used for actual length of the current value processed, what do you think?

Reply:
For example we have a method `lengthExceedsPrecision`, if we rename that var to `length` it will be confusing.